freqtrade/strategy/interface.py

```diff
@@ -707,35 +717,39 @@ def _analyze_ticker_internal(self, dataframe: DataFrame, metadata: dict) -> Data
         # always run if process_only_new_candles is set to false
         if (not self.process_only_new_candles or
                 self._last_candle_seen_per_pair.get(pair, None) != dataframe.iloc[-1]['date']):
+
             # Defs that only make change on new candle data.
             dataframe = self.analyze_ticker(dataframe, metadata)
+
             self._last_candle_seen_per_pair[pair] = dataframe.iloc[-1]['date']
-            self.dp._set_cached_df(
-                pair, self.timeframe, dataframe,
-                candle_type=self.config.get('candle_type_def', CandleType.SPOT))
+
+            candle_type = self.config.get('candle_type_def', CandleType.SPOT)
+            self.dp._set_cached_df(pair, self.timeframe, dataframe, candle_type=candle_type)
+            self.dp.emit_df((pair, self.timeframe, candle_type), dataframe)
+
         else:
             logger.debug("Skipping TA Analysis for already analyzed candle")
-            dataframe[SignalType.ENTER_LONG.value] = 0
-            dataframe[SignalType.EXIT_LONG.value] = 0
-            dataframe[SignalType.ENTER_SHORT.value] = 0
-            dataframe[SignalType.EXIT_SHORT.value] = 0
-            dataframe[SignalTagType.ENTER_TAG.value] = None
-            dataframe[SignalTagType.EXIT_TAG.value] = None
+            dataframe = remove_entry_exit_signals(dataframe)
```

Comment:
I dont understand why this is here, and I don't understand how the `if` `else` logic is leading to this. I thought this should be based on the user config `external_message_consumer: {"remove_entry_exit_signals": false}`?

Reply:
This was happening already before any of these changes were made. It's there because the conditional is looking to see if there is a new candle on the dataframe that we haven't seen. If there isn't, it means we've already seen all of those candles so far, and have used all of the signals on those candles. This part just removed any left over signals so we weren't constantly acting on the same ones over and over. But since the functionality that was doing that was being reused, I moved it into a function and made use of the function in the ExternalMessageConsumer as well.

Reply:
The remove_entry_exit_signals for the ExternalMessageConsumer is on line 309